Minio/V4Authenticator.cs

```diff
@@ -110,35 +111,51 @@ public string Authenticate(HttpRequestMessageBuilder requestBuilder)
             }
             this.SetDateHeader(requestBuilder, signingDate);
             this.SetSessionTokenHeader(requestBuilder, this.sessionToken);
+
             SortedDictionary<string, string> headersToSign = this.GetHeadersToSign(requestBuilder);
             string signedHeaders = this.GetSignedHeaders(headersToSign);
-            string canonicalRequest = this.GetCanonicalRequest(requestBuilder, headersToSign);
-            byte[] canonicalRequestBytes = System.Text.Encoding.UTF8.GetBytes(canonicalRequest);
-            string canonicalRequestHash = this.BytesToHex(this.ComputeSha256(canonicalRequestBytes));
-            string region = this.GetRegion(requestUri.Host);
-            string stringToSign = this.GetStringToSign(region, signingDate, canonicalRequestHash);
 
-            byte[] signingKey = this.GenerateSigningKey(region, signingDate);
-
-            byte[] stringToSignBytes = System.Text.Encoding.UTF8.GetBytes(stringToSign);
+            string canonicalRequest = this.GetCanonicalRequest(requestBuilder, headersToSign, assumeRole);
+            byte[] canonicalRequestBytes;
+            if (assumeRole)
+            {
+                canonicalRequestBytes = Encoding.ASCII.GetBytes(canonicalRequest);
+            }
+            else
+            {
+                canonicalRequestBytes = Encoding.UTF8.GetBytes(canonicalRequest);
+            }
```

Comment:
this doesn't look correct @ebozduman - why do you need UTF8 v/s ASCII ?

Reply:
@harshavardhana 

I agree. This is where I actually had a "To be discussed during review" comment, but during the debug/comment cleanup before my final push, it looks like I accidentally wiped it out. 

The original code was using UTF8 encoding. The AssumeRole test I used during the development kept failing at this point. I believe I checked on the server side and I thought the problem is the encoding type and when I tested with ASCII encoding, it worked. It did not make sense, so I decided to put a comment here so it can be discussed during the code review. 

What I will do is to remove the ASCII encoding and run the test to see if I can reproduce the issue.
Please let me know if you have something else to recommend.